src/context/dropdown-context.js

```diff
@@ -0,0 +1,118 @@
+import { createContext, useContext, useState, useEffect } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { v4 as uuid } from "uuid";
+import { useLike } from "../context";
+import {
+  WatchLaterIcon,
+  OutlinedWatchLaterIcon,
+  PlaylistIcon,
+  TrashIcon,
+  LikeIcon,
+  OutlinedLikeIcon,
+} from "../assets";
+
+const DropdownContext = createContext();
+
+const DropdownProvider = ({ children }) => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { addToLikedVideos, removeFromLikedVideos } = useLike();
+  const [showDropdown, setShowDropdown] = useState("");
+
+  useEffect(() => {
+    const handleDocumentClick = () => {
+      if (showDropdown !== "") setShowDropdown("");
+    };
+
+    document.addEventListener("click", handleDocumentClick);
+
+    return () => {
+      document.removeEventListener("click", handleDocumentClick);
+    };
+  }, [showDropdown]);
+
+  const getDropdownList = (isInWatchLater, isInLiked) => {
+    // Home page dropdownlist
+    let dropdownList = [
+      {
+        _id: uuid(),
+        option: `${isInWatchLater ? "Remove from" : "Add to"} Watch Later`,
+        Icon: () =>
+          isInWatchLater ? (
+            <WatchLaterIcon size={22} className="mr-0p5" />
+          ) : (
+            <OutlinedWatchLaterIcon size={22} className="mr-0p5" />
+          ),
+      },
+      {
+        _id: uuid(),
+        option: "Save to Playlist",
+        Icon: () => <PlaylistIcon size={24} className="mr-0p5" />,
+      },
+    ];
+
+    //History page dropdown
+    dropdownList =
+      location.pathname === "/history"
+        ? [
+            {
+              _id: uuid(),
+              option: "Remove from History",
+              Icon: () => <TrashIcon size={24} className="mr-0p5" />,
+            },
+          ]
+        : dropdownList;
+
+    // dropdown for pages other than home and history
```

Comment:
I liked the comments they some how increase the readability and understandability of the code